src/dbnode/server/server.go

```diff
@@ -371,6 +374,16 @@ func Run(runOpts RunOptions) {
 		logger.Warn("max index query IDs concurrency was not set, falling back to default value")
 	}
 
+	// Set global index options.
+	if n := cfg.Index.RegexpDFALimitOrDefault(); n > 0 {
+		regexp.SetStateLimit(n)
+		levenshtein.SetStateLimit(n)
+		levenshtein2.SetStateLimit(n)
+	}
+	if n := cfg.Index.RegexpFSALimitOrDefault(); n > 0 {
+		regexp.SetDefaultLimit(n)
```

Comment:
This fails to compile because `n` must be `uint`.
Also, this is somewhat suspicious as it changes the default value (which is a global variable), do we really need to override the **default** from the config?